src/tests/compiler/ast/doc/jsDocTypeTagTests.ts

```diff
@@ -0,0 +1,23 @@
+import {expect} from "chai";
+import {TypeGuards, JSDocTypeTag} from '../../../../main';
+import {getInfoFromText} from '../../testHelpers';
+
+describe.only(nameof(JSDocTypeTag), () => {
+    function getInfo(text: string) {
+        const info = getInfoFromText(text);
+        return { descendant: info.sourceFile.getFirstDescendantOrThrow(TypeGuards.isJSDocTypeTag), ...info };
+    }
+
+    describe(nameof<JSDocTypeTag>(d => d.getTypeExpression), () => {
+        it("returns undefined when typeExpression is undefined", () => {
+            const { descendant } = getInfo("/** @type */\nvar bar = 1;");
+            descendant.compilerNode.typeExpression = undefined;
```

Comment:
Not happy about this. The Typescript type definition says it can be undefined. But I can't seem to find any way to get it actually undefined. Doesn't matter what I put behind the "@type" tag, it will always give back a value, even when it's empty.

I've put it as undefined here myself to check show that the function doesn't do anything weird when its undefined, but I don't think it can ever be undefined. Seems like this is a mistake in Typescript.

Reply:
Yeah, most likely a mistake in the compiler.... that occasionally happens. Thanks for calling attention to this... I've made it return undefined if the node length is zero.